src/vs/platform/tunnel/node/tunnelService.ts

```diff
@@ -84,7 +84,9 @@ class NodeRemoteTunnel extends Disposable implements RemoteTunnel {
 
 	public async waitForReady(): Promise<this> {
 		// try to get the same port number as the remote port number...
-		let localPort = await findFreePortFaster(this.suggestedLocalPort ?? this.tunnelRemotePort, 2, 1000);
+		const startPort = this.suggestedLocalPort ?? this.tunnelRemotePort;
+		const hostname = isMacintosh && startPort < 1024 && !isPortPrivileged(startPort, this.defaultTunnelHost, OS, os.release()) ? '0.0.0.0' : '127.0.0.1';
```

Comment:
We shouldn't default to all interfaces unless the user has explicitly asked us to by using the setting.

Reply:
done :+1: